backend/src/permissions/permissions.service.spec.ts

```diff
@@ -46,13 +46,31 @@ import { NoteUserPermission } from './note-user-permission.entity';
 import { PermissionsModule } from './permissions.module';
 import { PermissionsService } from './permissions.service';
 
+function mockedEventEmitter(eventEmitter: EventEmitter2) {
+  return jest.spyOn(eventEmitter, 'emit').mockImplementationOnce((event) => {
+    expect(event).toEqual(NoteEvent.PERMISSION_CHANGE);
+    return true;
+  });
+}
+
+function mockNoteRepo(noteRepo: Repository<Note>) {
+  jest
+    .spyOn(noteRepo, 'save')
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    .mockImplementationOnce(async (entry: Note) => {
+      return entry;
+    });
+}
+
 describe('PermissionsService', () => {
   let service: PermissionsService;
   let notes: Note[];
   let noteRepo: Repository<Note>;
   let userRepo: Repository<User>;
   let groupRepo: Repository<Group>;
   let eventEmitter: EventEmitter2;
+  let emitEvent: jest.SpyInstance;
```

Comment:
The name isn't quite fitting. The variable doesn't contain an Event.
Something like `eventEmitterEmitSpy` would be more precise.

Reply:
done